internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/ssl/SslInitializer.java

```diff
@@ -323,12 +326,20 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
           } catch (Throwable ex) {
             LOG.warn("Unable to obtain remote CN for {}", ctx.channel().remoteAddress(), ex);
           }
-          LOG.info(
-              "SSL Handshake with {} ({}) {} in {} ms.",
-              ctx.channel().remoteAddress(),
-              remoteCN != null ? remoteCN : "unknown principal",
-              succeed ? "succeeded" : "failed",
-              (Time.nanoTime() - ctx.channel().attr(SSL_HANDSHAKE_START_TS).getAndSet(null)) / 1000000);
+          StringBuilder logMessage = new StringBuilder();
+          logMessage.append("SSL Handshake with ")
+              .append(ctx.channel().remoteAddress())
+              .append(" (")
+              .append(remoteCN != null ? remoteCN : "unknown principal")
+              .append(") ")
+              .append(succeed ? "succeeded" : "failed")
```

Comment:
can we plug in the failed caused without adding another if statement later?

Reply:
Good call! Thanks.